android/app/build.gradle

```diff
@@ -191,7 +192,7 @@ android {
             def abi = output.getFilter(OutputFile.ABI)
             if (abi != null) {  // null for the universal-debug, universal-release variants
                 output.versionCodeOverride =
-                        versionCodes.get(abi) * 1048576 + defaultConfig.versionCode
+                        defaultConfig.versionCode * 1000 + versionCodes.get(abi)
```

Comment:
Could this be problematic for any existing installations (newer builds will have lower version codes)? Assuming we don't have any, so this isn't an issue?

Reply:
Oh i'm not really sure of the side effects. I'm gonna merge this and then do some testing.

I am really not sure if we are going to be compatible with past versions. I would have thought not. I think it might be some not insignificant changes and migration needed?

Reply:
@wlthomson I remember having the same reservation when I read this, though I'd commented 😶 . Pretty sure mobile equivalent munges the package.json semver into a number where major>minor>patch, allowing a few digits for each (or at least it used to). That said, not familiar that the versioning is adjusted for different instruction sets. If a subsequent release's version is lower than the current, I'm pretty sure you won't be able to publish it or install it or upgrade the app on device.

So definitely worth checking for mSupply ColdChain v0.3!

The implementation needs to be able to preserve existing data; the app is deployed for 2 customers (at least?) and they will not appreciate their datalogs vanishing - especially if they haven't diligently downloaded/emailed them to themselves for backup.